src/python/pants/BUILD

```diff
@@ -40,6 +40,16 @@ python_distribution(
     entry_points={"console_scripts": {"pants": "src/python/pants/bin:pants"}},
 )
 
+pex_binary(
+    name="pants-pex",
+    dependencies=[":pants-packaged"],
+    script="pants",
+    execution_mode="venv",
+    shebang="/usr/bin/env python",
+    strip_pex_env=False,
+    layout="zipapp",
+)
```

Comment:
This was copied from `release.py`, but without the `--platforms` flag. As I understand it, this will resolve only for the local platform, so that's exactly what we want!

Reply:
I also am hardcoding `zipapp` PEX to guard against `__defaults__` accidentally changing that.

Reply:
I believe that you understand correctly.